game/src/camera.rs

```diff
@@ -79,17 +106,103 @@ fn manage_camera_projection(// mut q_cam: Query<&mut OrthographicProjection, Wit
     // TODO
 }
 
-//TODO
-fn camera_follow_player(
-    mut q_cam: Query<&mut Transform, With<MainCamera>>,
+/// Updates the Camera rig (ie, the camera target) based on where the player is going.
+fn camera_rig_follow_player(
+    mut rig: ResMut<CameraRig>,
     q_player: Query<&Transform, (With<PlayerGent>, Without<MainCamera>)>,
+    // Keeps track if the camera is leading ahead, or behind the player
+    mut lead_bckwrd: Local<bool>,
```

Comment:
name is weird but this way its default (false) is the one we want without needing an extra check in the function.